sstables/storage.cc

```diff
@@ -598,7 +605,10 @@ future<> init_table_storage(const data_dictionary::storage_options& so, sstring
         },
         [] (const data_dictionary::storage_options::s3&) -> future<> {
             co_return;
-        }
+        },
+        [&dir] (const data_dictionary::storage_options::mirrored&) -> future<> {
+            co_await io_check([&dir] { return recursive_touch_directory(dir); });
```

Comment:
Local filestystem part has more complex initialization, it should create staging/ and upload/ subdirs. I don't think upload/ makes sense in the mirrored case, but missing staging/ should come back to bite as later -- IIRC when moving sstable to staging/ this directory is assumed to pre-exist and open(O_CREAT)-ing a file in it would fail if this assumption happens not to be true